Fix ContentSingle play button and loading state

rkdavidson opened the pull request:
# About

Fixes #150 

Adds a loading state to `ContentSingle`, and cleans up the `videos` handling accordingly.
This fixes the problem of #150 and a previously unnoticed merge quirk that caused duplicate props.

Another sneaky subtle problem fixed while in here was that the `seo` data for `video` was always referencing the first video. I made it reference `currentVideo` instead, though I don't have a firm grasp on what the value/effect of dynamically changing `<meta>` tags on the page like that is 🤔.
It shouldn't _hurt_, but it probably doesn't help 🤷‍♂️.


# Testing
Load a piece a **Message** content item (go to Discover and click one from the **Messages** section).
Verify there's a brief loading state, then that the page loads properly with a video play icon on the hero image.
Tap play, wait a few seconds, reload the page, and verify the same thing happens. Page should load and show the play button again.

asleepingpanda commented:
@rkdavidson The build is failing in netlify and I'm not sure why. I'm gonna pull it first thing in the morning and look into it a bit

rkdavidson commented:
@asleepingpanda Ah, I see why. All the `Single` pages dynamically derived from the new routing scheme will need to handle a `loading` state. Will update tonight or tomorrow morning.

rkdavidson commented:
Hey @asleepingpanda I think I fixed the build issues. All the components who received `ContentItemProvider` data assumed that the data would be there at initial render, since we were previously fetching things server side.

Since it's a temporary client side lookup, we needed to handle loading states.
I added that to `EventSingle`, `PageSingle`, and `LocationSingle` too.
